runtime/tests/Test_NullQubit.cpp

```diff
@@ -897,30 +896,23 @@ TEST_CASE("Test NullQubit device shots methods", "[NullQubit]")
     }
 }
 
-TEST_CASE("Test NullQubit device resource tracking", "[NullQubit]")
+TEST_CASE("Test NullQubit device resource tracking integration", "[NullQubit]")
 {
     // The name of the file where the resource usage data is stored
     constexpr char RESOURCES_FNAME[] = "__pennylane_resources_data.json";
 
-    // Open a file for writing the resources JSON
-    FILE *resource_file_w = fopen(RESOURCES_FNAME, "wx");
-    if (resource_file_w == nullptr) {                            // LCOV_EXCL_LINE
-        FAIL("Failed to open resource usage file for writing."); // LCOV_EXCL_LINE
-    }
-
     std::unique_ptr<NullQubit> dummy = std::make_unique<NullQubit>();
     CHECK(dummy->IsTrackingResources() == false);
 
     std::unique_ptr<NullQubit> sim =
-        std::make_unique<NullQubit>("{'track_resources':True}", RESOURCES_FNAME);
+        std::make_unique<NullQubit>("{'track_resources':True, 'resources_fname':'" +
+                                    std::string(RESOURCES_FNAME) + ", 'compute_depth':True'}");
```

Comment:
Did the quotation marks get messed up here?
```suggestion
        std::make_unique<NullQubit>("{'track_resources':True, 'resources_fname':'" +
                                    std::string(RESOURCES_FNAME) + "', 'compute_depth':True}");
```